packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -985,6 +999,13 @@ public void onFailure(@NonNull Exception e) {
       this.exception = null;
     }
   }
+
+  private void removeSnapshotListeners() {
+    for (int i = 0; i < listenerRegistrations.size(); i++) {
+      listenerRegistrations.get(i).remove();
```

Comment:
This must use `valueAt(i)`, as listenerRegistrations in a SparseArray (a map-like type) and the key is the handle ID.

Have tested it locally just now and ran into NPE using `.get(i)`.

Reply:
Thanks for flagging that!